### DISCUSSION TITLE
corExp

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#142] [General]
Hello,

I'm trying to get a spatial correlation structure corExp to run in gllvm. Is there somewhere an example?

Otherwise......something like this?
gllvm(......,
studyDesign = GroupStruc,
dist = CoordData,
row.eff = ~ corExp(Xkm + Ykm | Group),
...

where CoordData is a matrix with two columns (Xkm and Ykm)?
And GroupStruc contains a factor  Group with only 1 level....like in glmmTMB?
Or do I need to make a 'pos' variable as in the glmmTMB page below?

The gllvm code complains that it cannot find Xkm (yet it is in CoordData).

Is the expression for corExp in gllvm the same as at the very top of this page (spatial exponential):
https://cran.r-project.org/web/packages/glmmTMB/vignettes/covstruct.html#spatial-correlations

Kind regards,

Alain

**Comment by BertvanderVeen** [comment]
There is no vignette yet, but @JenniNiku is working on it. The way to do it is:

```
gllvm(......,
studyDesign = GroupStruc,
dist = distances,
row.eff = ~ corExp(1 | Group)
```

where "group" are the group numbers and "distances" are (euclidean) distances between the coordinates of the group units. So, with the spider dataset (which has no coordinates, but because everyone just loves that so much):

```
model <- gllvm(spider$abund, row.eff = ~(1|site), studyDesign = data.frame(site=1:28), family = "poisson", dist = diag(28))
```
 although, of course, "site" can be any type of grouping and "dist" should be a matrix of distances between the groups/sites. Note that when my new branch is pulled into gllvm, "dist" becomes a list, facilitating the (potential) inclusion of multiple spatial effects at different levels (if that is a thing people do..).

**Reply by BertvanderVeen** [reply]
Woops, I am misinformed. You are correct that "dist" should include the x-y coordinates of your groups.

Bert

**Reply by AlainZuur** [reply]
Thanks....that runs. Will play around with it!

Kind regards,

Alain
